com.unity.render-pipelines.high-definition/Runtime/Material/Decal/DecalUtilities.hlsl

```diff
@@ -261,3 +261,21 @@ DecalSurfaceData GetDecalSurfaceData(PositionInputs posInput, float3 vtxNormal,
 
     return decalSurfaceData;
 }
+
+DecalSurfaceData GetDecalSurfaceData(PositionInputs posInput, FragInputs input, inout float alpha)
+{
+    float3 vtxNormal = input.tangentToWorld[2];
+    DecalSurfaceData decalSurfaceData = GetDecalSurfaceData(posInput, vtxNormal, alpha);
+
+#ifdef _DOUBLESIDED_ON
+    if (!input.isFrontFace)
+    {
+        if (_DoubleSidedConstants.x == -1.0f) // Flip mode
+            decalSurfaceData.normalWS.xyz *= -1.0f;
+        else if (_DoubleSidedConstants.z == -1.0f) // Mirror mode
+            decalSurfaceData.normalWS.z *= -1.0f;
```

Comment:
in the end it was quite simple

Reply:
indeed :)

can you rewrite the code like this:

    // 'doubleSidedConstants' is float3(-1, -1, -1) in flip mode and float3(1, 1, -1) in mirror mode.
    // It's float3(1, 1, 1) in the none mode.
    float3 flipSign = input.isFrontFace ? float3(1.0, 1.0, 1.0) : doubleSidedConstants;
    decalSurfaceData.normalWS *= flipSign;

to remove the various branch